src/Models/Traits/Searchable.php

```diff
@@ -2,42 +2,24 @@
 
 namespace Rapidez\Core\Models\Traits;
 
-use Exception;
-use Illuminate\Database\Eloquent\Builder;
-use Illuminate\Database\Eloquent\Collection;
-use Illuminate\Database\Eloquent\Model;
 use Laravel\Scout\Searchable as ScoutSearchable;
 use TorMorten\Eventy\Facades\Eventy;
 
 trait Searchable
 {
     use ScoutSearchable;
 
+    /**
+     * {@inheritdoc}
+     */
     public function toSearchableArray(): array
     {
-        $data = $this->searchableData($this->toArray(), $this);
-
-        // TODO: Is this filter still useful? Overriding the model gives you full control
```

Comment:
Overriding the model should be used in projects. However packages do not have this luxury.
This is why i believe it's better to keep this.